radix-engine-tests/tests/system/toolkit_receipt.rs

```diff
@@ -0,0 +1,738 @@
+#![cfg(feature = "std")]
+
+use std::path::PathBuf;
+
+use radix_engine::system::system_modules::execution_trace::{ResourceSpecifier, WorktopChange};
+use radix_engine_interface::prelude::MetadataValue;
+use radix_engine_tests::*;
+use radix_engine_toolkit::receipt::*;
+use radix_transaction_scenarios::executor::*;
+use scrypto::*;
+use scrypto_test::prelude::*;
+
+#[test]
+fn test_toolkit_receipt_roundtrip_property() {
```

Comment:
I wonder if these should be in the `radix-engine-toolkit` crate?

Although happy for them to be here if you'd prefer.

If we move them to the `radix-engine-toolkit`, we'll need to update `ci.yml` to include such a test.

Reply:
(maybe not worth doing else we'll get weird merge conflicts upon merging with develop)

Reply:
I can see them being in either crate. It seems somewhat nice to me to have them in the `radix-engine-tests` crate, but they also don't quite fit in with the engine abstractions. I have a slight preference for keeping them here but could be argued either way. 

If we plan on moving them then I'd say let's move them on `develop` and not on this branch in order to avoid merge conflicts.